subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/BeanNode.java

```diff
@@ -17,6 +17,7 @@
 package org.gradle.api.internal.tasks.properties;
 
 public interface BeanNode<SELF extends BeanNode<SELF>> {
-    boolean isIterable();
+    boolean isRoot();
     Iterable<SELF> asIterable(NestedBeanContext<SELF> context);
```

Comment:
There is no need for an `Iterable` here, as we will instantly ask for its `Iterator`. It would be better to directly implement `Iterator<SELF> getIterator(...)` or something here.

Reply:
`getIterator` it is.